src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1169,11 +1177,19 @@ private bool IsDefaultLiteralAllowedInBinaryOperator(BinaryOperatorKind kind, Bo
                 resultKind = possiblyBest.HasValue ? LookupResultKind.Viable : LookupResultKind.Empty;
             }
 
-            if (possiblyBest.HasValue &&
-                (object)possiblyBest.Signature.Method != null)
+            if (possiblyBest.HasValue)
             {
-                Symbol symbol = possiblyBest.Signature.Method;
-                ReportDiagnosticsIfObsolete(diagnostics, symbol, node, hasBaseReceiver: false);
+                MethodSymbol operatorMethod = possiblyBest.Signature.Method;
+                if ((object)operatorMethod != null)
+                {
+                    ReportDiagnosticsIfObsolete(diagnostics, operatorMethod, node, hasBaseReceiver: false);
+
+                    if (operatorMethod.ContainingType.IsInterface &&
+                        operatorMethod.ContainingModule != Compilation.SourceModule)
+                    {
+                        Binder.CheckFeatureAvailability(node, MessageID.IDS_DefaultInterfaceImplementation, diagnostics);
+                    }
+                }
```

Comment:
Consider extracting a helper method for use here and in `BinaryOperatorOverloadResolution`.